src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -23,7 +23,9 @@ public SocketAsyncContext AsyncContext
         {
             get
             {
-                return _innerSocket.AsyncContext;
+                return _innerSocket == null ?
+                    SocketAsyncContext.ClosedAsyncContext :
+                    _innerSocket.AsyncContext;
```

Comment:
Are there any race conditions we need to be concerned about, with _innerSocket being set from non-null to null concurrently with this call?  Or is concurrent usage not allowed, or _innerSocket never goes back to null after being non-null?